app/helpers/tasklist_helper.rb

```diff
@@ -29,7 +29,7 @@ def first_page_or_check_answers(subsection, case_log)
     path = if subsection.is_started?(case_log)
              "case_log_#{subsection.id}_check_answers_path"
            else
-             "case_log_#{subsection.pages.first.id}_path"
+             "case_log_#{subsection.applicable_questions(case_log).first.page.id}_path"
```

Comment:
Should there be a test for the case this change is covering? Or have I missed one?

Reply:
I think this was originally covered by the tests added in this PR - I did recently have to rebase it and update the tests due to all the changes that happened while I was away. It's not breaking any existing tests at the moment so I'd be willing to say that's a promising sign

Reply:
Might be worth adding a unit test for the specific edge case here https://github.com/communitiesuk/mhclg-data-collection-beta/blob/main/spec/helpers/tasklist_helper_spec.rb#L44 but not a blocker.